index.js

```diff
@@ -33,9 +39,9 @@ module.exports = function(glob, options) {
 
   // make glob absolute
   if (rootDir && glob.charAt(0) === '/') {
-    glob = path.join(rootDir, glob);
-  } else if (!isAbsolute(glob)) {
-    glob = path.resolve(cwd, glob);
+    glob = join(rootDir, glob);
+  } else if (process.platform === 'win32' || !isAbsolute(glob)) {
```

Comment:
> This is solved by removing the process.platform === 'win32' check. The !isAbsolute call is enough to determine if this branch is needed.

@phated I removed this and one of the tests (in `to-absolute-glob`) fails on windows because of this pattern: `\\!(foo)`.

The `\\` is in there because it's an escape character but `isAbsolute` will see it as the start of an absolute path in Windows.

I fixed this by checking if the glob starts with an escape sequence.

Going through all this has helped me understand file paths a lot more and hopefully I'll be able to contribute to the glob-spec.